scripts/process_licenses.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python
+
+# This script processes licenses of 3rd party dependencies and stores them in the JAR. The rules are:
+# 1. Dependencies, which contains a license file should be put into the shaded JAR as-is.
+# 2. Dependencies, which do not contain a license file should be mentioned in the file ADDITIONAL_LICENCES, together with the name of its license.
+#
+#
+# The script accepts the following arguments:
+# * DEPENDENCY_LIST_FILE_PATH
+#     * Can be obtained by running mvn dependency:list -DincludeScope=runtime -DoutputFile=target/dependency_list.txt
+# * DEPENDENCIES_DIR
+#     * Directory containging the JAR files of all SDK dependencies. Automatically generated by `mvn clean package` in target/dependency-jars
+# * TARGET_DIR
+#     * Where to save all output, should be target/generated-sources/META-INF/third-party-licenses
+#
+#
+# Useful mvn commands:
+# * mvn clean license:add-third-party
+#     * Generate dependency report; useful to find out licenses for dependencies that don't ship with a license file
+# * mvn dependency:list -DincludeScope=runtime -DoutputFile=target/dependency_list.txt
+#     * Used as input of this script (DEPENDENCY_LIST_FILE_PATH)
+import sys
+from pathlib import Path
+from zipfile import ZipFile
+
+# License name constants
+APACHE_LICENSE = "Apache License 2.0"
+BSD_2_CLAUSE_LICENSE = "2-Clause BSD License"
+BSD_3_CLAUSE_LICENSE = "3-Clause BSD License"
+EDL_10_LICENSE = "EDL 1.0"
+MIT_LICENSE = "The MIT License"
+GO_LICENSE = "The Go license"
+BOUNCY_CASTLE_LICENSE = "Bouncy Castle License"
+
+# The SDK does not need to include licenses of dependencies, which aren't shaded
+IGNORED_DEPENDENCIES = {"net.snowflake:snowflake-jdbc", "org.slf4j:slf4j-api"}
+
+# List of dependencies, which don't ship with a license file.
+# Only add a new record here after verifying that the dependency JAR does not contain a license!
```

Comment:
might be worth calling it out in README?

Reply:
Added README section about third-parties